### PR TITLE
Support success callbacks

### PR DESCRIPTION
**Pull request by mgregson**
Extending #217 to provide callbacks for more scheduling functions. Also correcting an error in MongoDB response handling preventing user-provided callbacks from being called when the MongoDB operation encounters an error.

### CODE REVIEW

**Comment by loris** [comment]
Great addition!

File: lib/agenda.js [hunk 1]
```diff
@@ -200,33 +200,33 @@ Agenda.prototype.every = function(interval, names, data) {
...
-    job.save();
+    job.save(cb);
     return job;
   }
 
-  function createJobs(when, names, data) {
+  function createJobs(when, names, data, cb) {
     return names.map(function (name) {
-      return createJob(when, name, data);
+      return createJob(when, name, data, cb);
```
Comment (Pyrolistical) [inline lib/agenda.js]: this is a bug and will cause cb to be called multiple times.

need to use `async.map` or similar

**Comment by Pyrolistical** [comment]
I need this, these methods should have never been implemented without callbacks or promises

**Comment by Pyrolistical** [comment]
Thank you @rschmukler for the quick release

**Comment by Zertz** [comment]
This is genius, thank you @Pyrolistical

File: lib/agenda.js (hunk 1 above)
Comment (mgregson) [inline lib/agenda.js]: You're right. I think this bug might be present in the code that mede it onto mainline too. Can you confirm?

Comment (rschmukler) [inline lib/agenda.js]: I fixed this before merging it into master. Thank you @mgregson

**Pull request closed without merging**